apps/ledger-live-mobile/src/families/solana/DelegationFlow/Summary.tsx

```diff
@@ -162,12 +162,22 @@ export default function DelegationSummary({ navigation, route }: Props) {
 
   const onContinue = useCallback(async () => {
     navigation.navigate(ScreenName.DelegationSelectDevice, {
+      ...route.params,
       accountId: account.id,
       parentId: parentAccount?.id,
       transaction: transaction as SolanaTransaction,
       status,
+      validatorName: chosenValidator?.name,
     });
-  }, [navigation, account.id, parentAccount?.id, transaction, status]);
+  }, [
+    route.params,
+    navigation,
+    account.id,
+    parentAccount?.id,
+    transaction,
+    status,
+    chosenValidator,
```

Comment:
I don't know if the `chosenValidator` obj is is memoised, but if not then the `chosenValidator?.name` string might be the more specific thing to watch in the dependency array than the whole object

Reply:
Thank you 👍 Will update!